modules/core/include/opencv2/core/cvdef.h

```diff
@@ -248,22 +248,18 @@ Cv64suf;
 #  define DISABLE_OPENCV_24_COMPATIBILITY
 #endif
 
-#if (defined _WIN32 || defined WINCE || defined __CYGWIN__) && defined CVAPI_EXPORTS
-#  define CV_EXPORTS __declspec(dllexport)
-#elif defined __GNUC__ && __GNUC__ >= 4
-#  define CV_EXPORTS __attribute__ ((visibility ("default")))
-#else
-#  define CV_EXPORTS
+#include "opencv2/cv_exports.h"
+
+// fix for Android FAT_JAVA_LIB: symbols should have default visibility even in static libraries
+#ifdef __ANDROID__
```

Comment:
Comment is not consistent with following code.
Android application native library should not export OpenCV API

Reply:
We build native static libraries and "fat Java lib" for Android in one build (`-DBUILD_SHARED_LIBS=OFF -DBUILD_FAT_JAVA_LIB=ON`). We need to export all symbols for the second artifact. Currently all symbols are exported in static builds on all platforms, this PR changes this behavior for all platforms except Android. I can only suggest reworking "fat Java lib" artifact, so that it will be produced same way as _opencv_world_ module (`-DBUILD_SHARED_LIBS=ON -DBUILD_opencv_world=ON -DBUILD_FAT_JAVA_LIB=ON`), but it cat take more time to investigate.

Reply:
Fat Java library is artifact for Android manager which is deprecated.
If application doesn't use Java API (doesn't require libopencv_java.so), so there is no reason to bundle whole OpenCV code into Android application.
More accurate way is no link with application JNI library with static OpenCV libraries - libopencv_core.a/etc. So we need to drop global exported symbols from these OpenCV static libraries.

So, there are 2 different builds:
- static libopencv_<>.a with global exported symbols => libopencv_java.so
- static libopencv_<>.a without global exported symbols and without any libopencv_java.so. Useful for using OpenCV API from native C++ code only (without any OpenCV Java bindings).